### DISCUSSION TITLE
Asking about the piano for clarification

### DISCUSSION THREAD

**Discussion by SirJustin90** [Monika-After-Story/MonikaModDev#9249] [QA - tech support + general] [answered]
So for the easter egg that "VGhlUG90YXRvR3V5IA==" added to the piano, I now lovingly call dpco, is there a way to enable it ourselves as most of it is coded (words and keys)? Would a file with the appropriate keys work or is it actually not fully coded and would need to be finished to actually work.

I just _slowly_ want to prep this if I can for when I unlock the piano and just want to know now that I understand the whole thing if we just need to play the notes shown, or if we actually need a file to play it, as it's already in the program. 

To clarify what I mean for anyone who is confused (SPOILER!!!):
aHR0cHM6Ly9naXRodWIuY29tL01vbmlrYS1BZnRlci1TdG9yeS9Nb25pa2FNb2REZXYvY29tbWl0LzlmNzQyZTY1MDI0YzJjMDc1NjQ2YWU3MzE0YWYyOGViZTNkNzI3MWQ=

**Comment by Wingdinggaster656** [comment]
You can add that song like any custom piano song.

The coding knowledge required is not too deep, and if you do a bit of research, you should be able to do it yourself.

(By the way it is very likely that Potato is just enjoying himself/herself/themselves. I don't really think this will be enabled at all)

**Comment by Justformas** [comment]
> To clarify what I mean for anyone who is confused (SPOILER!!!):
> aHR0cHM6Ly9naXRodWIuY29tL01vbmlrYS1BZnRlci1TdG9yeS9Nb25pa2FNb2REZXYvY29tbWl0LzlmNzQyZTY1MDI0YzJjMDc1NjQ2YWU3MzE0YWYyOGViZTNkNzI3MWQ=

How is that really any less confusing? Just search in zz_pianokeys.rpy for either "d__p_c__o.json" or line 1538 where the song starts.

> (By the way it is very likely that Potato is just enjoying himself/herself/themselves. I don't really think this will be enabled at all)

I wonder if it would be allowed for someone to submit a json in piano notes for this.
Actually that could help accomplish line 1934: # TODO: need to finish dpco one day

**Comment by SirJustin90** [comment]
> How is that really any less confusing? Just search in zz_pianokeys.rpy for either "d__p_c__o.json" or line 1538 where the song starts.
> 
> > (By the way it is very likely that Potato is just enjoying himself/herself/themselves. I don't really think this will be enabled at all)
> 
> I wonder if it would be allowed for someone to submit a json in piano notes for this. Actually that could help accomplish line 1934: # TODO: need to finish dpco one day

That's what I was wondering. Also that is exactly what I was referencing, but as it is meant to be a secret I made it a little harder to find on purpose.

Same reason I didn't saw the user who made it out loud either haha.

But you both knew what I meant anyways.

If you are familiar with DDLC the info I posted should be easy to decode.

**Comment by Justformas** [comment]
> That's what I was wondering. Also that is exactly what I was referencing, but as it is meant to be a secret I made it a little harder to find on purpose.
> 
> Same reason I didn't saw the user who made it out loud either haha.
> 
> But you both knew what I meant anyways.
> 
> If you are familiar with DDLC the info I posted should be easy to decode.

So in other words: "spoilers if you can solve/decode this puzzle".
I actually didn't know this was that Base64 stuff, but I found something despite that. Not the actual commit from 2018 though which I now see more details in; I found traces of it in the actual master file of zz_pianokeys.rpy and the line numbers I was referring to are for that file.

I personally really don't like those types of puzzles that you need to translate, or that require a tool outside of the game. At the very least I'd want a hint about that like you gave me. But out of curiosity I also tried decoding in that manner for Monika's character file that gets generated when you take her out somewhere; to see if there was hidden messages that get generated. But as I expected, nothing.

**Comment by ThePotatoGuy** [comment, marked as answer]
It's unreleased because I only completed the first verse and chorus 😄 I'm not a musician nor spanish speaker so doing just that took way longer than I would like, and the original intent was to showcase a new debug feature for piano song makers. (See video below)

I'm probably not going to finish this, so if you can finish the full song and submit it here then we'll release it officially (or unofficially) as an actual playable song.


https://user-images.githubusercontent.com/3499462/172026495-2793e05b-7619-488b-8ed2-faf6e3178351.mp4

**Comment by Justformas** [comment]
> I'm not a musician nor spanish speaker

I'm not a spanish speaker either, and though I could just look the lyrics up to translate, I would like if there was a translation that could be shown in-game whether officially or unofficially.

**Comment by SirJustin90** [comment]
Actually there is already an English translation of it with "don't click this video if you like (song)" as it's title on YouTube etc. that is probably the most correct and appropriate translation I've seen to date for it.

**Comment by SirJustin90** [comment]
Also I am clueless beyond python, scripts and some c+/Java. So .json is a nightmare (the are converters I suppose).

However I may tinker and put some thought into that haha, perhaps others that already are more versed will as well.

**Comment by Justformas** [comment]
> Actually there is already an English translation of it with "don't click this video if you like (song)" as it's title on YouTube etc. that is probably the most correct and appropriate translation I've seen to date for it.

So it was already an actual song, and those were blanks to fill in. Never heard of it. Not the first time the playing field wasn't really fair to me, though Google basically did it for me. But fine I'll "play along". It's RGVzcGFjaXRv right?
